app/src/main/java/org/mozilla/focus/search/SearchEngineManager.java

```diff
@@ -157,6 +164,10 @@ public synchronized SearchEngine getDefaultSearchEngine(Context context) {
     }
 
     public void awaitLoadingSearchEnginesLocked() {
+        if (!loadHasBeenTriggered) {
+            throw new IllegalStateException("Attempting to retrieve search engines without a corresponding init()");
```

Comment:
We could trigger a load here - but I like your approach more. Throwing will uncover logical errors we made.